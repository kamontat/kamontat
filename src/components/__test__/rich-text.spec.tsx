import React from "react"
import { render } from "@testing-library/react"
import RichText from "../ContentfulRichText"

const mock = JSON.stringify({
  data: {},
  content: [
    {
      data: {},
      content: [
        {
          data: {},
          marks: [],
          value: "hello world",
          nodeType: "text",
        },
        { data: {}, marks: [{ type: "underline" }], value: "3", nodeType: "text" },
        {
          data: {},
          marks: [],
          value: " times. ",
          nodeType: "text",
        },
        { data: {}, marks: [{ type: "bold" }], value: "New word", nodeType: "text" },
        { data: {}, marks: [], value: " will create ", nodeType: "text" },
        { data: {}, marks: [{ type: "bold" }], value: "new name", nodeType: "text" },
        { data: {}, marks: [], value: ".", nodeType: "text" },
      ],
      nodeType: "paragraph",
    },
  ],
  nodeType: "document",
})

describe("Contentful.RichText", () => {
  it("generate empty component", () => {
    const { container } = render(<RichText />)
    expect(container.firstChild).toMatchSnapshot()
  })

  it("generate content from raw", () => {
    const { container } = render(<RichText raw={mock} />)
    expect(container.firstChild).toMatchSnapshot()
  })
  it("generate content from raw and references", () => {
    const { container } = render(<RichText raw={mock} references={[]} />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
