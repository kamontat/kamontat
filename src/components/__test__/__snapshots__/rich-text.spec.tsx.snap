// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contentful.RichText generate content from raw 1`] = `
.emotion-0 {
  font-size: 1rem;
  line-height: 1.625;
  white-space: pre-line;
}

.emotion-2 {
  font-weight: 700;
}

<p
  class="emotion-0 emotion-1"
>
  hello world
  <u>
    3
  </u>
   times. 
  <span
    class="emotion-2 emotion-3"
  >
    New word
  </span>
   will create 
  <span
    class="emotion-2 emotion-3"
  >
    new name
  </span>
  .
</p>
`;

exports[`Contentful.RichText generate content from raw and references 1`] = `
.emotion-0 {
  font-size: 1rem;
  line-height: 1.625;
  white-space: pre-line;
}

.emotion-2 {
  font-weight: 700;
}

<p
  class="emotion-0 emotion-1"
>
  hello world
  <u>
    3
  </u>
   times. 
  <span
    class="emotion-2 emotion-3"
  >
    New word
  </span>
   will create 
  <span
    class="emotion-2 emotion-3"
  >
    new name
  </span>
  .
</p>
`;

exports[`Contentful.RichText generate empty component 1`] = `<p />`;
